frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,219 @@
+  const [selected, setSelected] = useState<Rating>(null);
+  const [freeformFeedback, setFreeformFeedback] = useState<string>("");
+  const [error, setError] = useState<boolean>(false);
+  const [survey, setSurvey] = useState<IClutch.feedback.v1.ISurvey>({});
+  const [requestId, setRequestId] = useState<string>("");
+  const maxLength = 180;
+  const debounceTimer = 500;
+
+  const trimmed =
+    freeformFeedback.trim().length > maxLength
+      ? `${freeformFeedback.trim().substring(0, maxLength - 3)}...`
+      : freeformFeedback;
+
+  /** Property objects used to extend components and remove console warnings */
+  const AlertProps = {
+    iconMapping: {
+      info: <MuiSuccessIcon style={{ color: "#3548d4" }} />,
+    },
+    style: {
+      margin: "32px",
+      alignItems: "center",
+    },
+  };
+
+  const textFieldProps = {
+    fullWidth: true,
+    InputProps: {
+      rows: 3,
+      rowsMax: 3,
+    },
+  };
+
+  // Will fetch the survey results for the given origin on load
+  React.useEffect(() => {
+    let data: IClutch.feedback.v1.ISurvey = defaults;
+
+    client
+      .post("/v1/feedback/getSurveys", {
+        origins: [opts.origin],
+      })
+      .then(response => {
+        const surveyData: IClutch.feedback.v1.IGetSurveysResponse = response?.data?.originSurvey;
+
+        data = surveyData[opts.origin] ?? defaults;
+      })
+      .catch((err: ClutchError) => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      })
+      .finally(() => {
+        setRequestId(uuid());
+        setSurvey(data);
+      });
+  }, []);
+
+  // Will debounce feedback requests to the server in case of multiple quick changes to selected
+  const sendFeedback = React.useCallback(
+    debounce((formData: IClutch.feedback.v1.ISubmitFeedbackRequest) => {
+      client
+        .post("/v1/feedback/submitFeedback", { userId: userId(), ...formData })
+        .catch((err: ClutchError) => {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        });
+    }, debounceTimer),
+    []
+  );
+
+  // On a change to submit or selected will attempt to send a feedback request
+  React.useEffect(() => {
+    if (selected) {
+      sendFeedback({
+        id: requestId,
+        feedback: {
+          ratingLabel: selected.label,
+          ratingScale: {
+            emoji: IClutch.feedback.v1.EmojiRating[selected.emoji],
+          },
+          urlPath: window.location.pathname,
+          freeformResponse: trimmed,
+        },
+        metadata: {
+          origin: IClutch.feedback.v1.Origin[opts.origin],
+          userSubmitted: hasSubmit,
+          survey,
+        },
+      });
+    }
+  }, [selected, hasSubmit]);
+
+  // Form onSubmit handler
+  const submitFeedback = e => {
+    if (e) {
+      e.preventDefault();
+    }
+    setSubmit(true);
+    if (opts.onSubmit) {
+      opts.onSubmit(true);
+    }
+  };
+
+  return (
+    <>
```

Comment:
similar question as above in regards to if this fragment is required

Reply:
I'll break it apart to return a separate component on submit which will get rid of the fragment and clean up some of the nesting.